CRM/Member/Form/MembershipView.php

```diff
@@ -154,8 +154,22 @@ public function preProcess() {
     $this->assign('context', $context);
 
     if ($this->membershipID) {
-      $params = ['id' => $this->membershipID];
-      CRM_Member_BAO_Membership::retrieve($params, $values);
+      $values = \Civi\Api4\Membership::get()
+        ->addSelect('*', 'status_id:label', 'membership_type_id:label', 'membership_type_id.financial_type_id', 'status_id.is_current_member')
+        ->addWhere('id', '=', $this->membershipID)
+        ->execute()
+        ->first();
+
+      // Ensure keys expected by MembershipView.tpl are set correctly
+      // Some of these defaults are overwritten dependant on context below
+      $values['financialTypeId'] = $values['membership_type_id.financial_type_id'];
+      $values['membership_type'] = $values['membership_type_id:label'];
+      $values['status'] = $values['status_id:label'];
+      $values['active'] = $values['status_id.is_current_member'];
+      $values['owner_contact_id'] = FALSE;
+      $values['owner_display_name'] = FALSE;
+      $values['campaign'] = FALSE;
+
       if (CRM_Financial_BAO_FinancialType::isACLFinancialTypeStatus()) {
         $finTypeId = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $values['membership_type_id'], 'financial_type_id');
         $finType = CRM_Contribute_PseudoConstant::financialType($finTypeId);
```

Comment:
One question maybe @eileenmcnaughton can answer - this financial ACL check may be redundant now if we are loading the membership via APIv4 with `checkPermissions = TRUE`. If so we could simplify it to see if the record was returned or not, like
```
if (!$values) {
  \CRM_Utils_System::permissionDenied();
}

Reply:
@colemanw yep - I'd like to connect with you on fixing up the membership acls for financialacls & then rip that out. There is an issue with merge-subquery I'd like to work through with you #22499 is the ref point but let's chat on chat so I can explain